chains/evm/deployment/utils/changesets/changesets.go

```diff
@@ -19,33 +18,40 @@ type NewFromOnChainSequenceParams[IN any, DEP any, CFG any] struct {
 	ResolveInput func(e deployment.Environment, cfg CFG) (IN, error)
 	// ResolveDeps resolves the dependencies for the sequence based on the environment and changeset config.
 	ResolveDep func(e deployment.Environment, cfg CFG) (DEP, error)
+	// ResolveMCMS resolves the MCMS configuration based on the environment and changeset config.
+	ResolveMCMS func(e deployment.Environment, cfg CFG) (MCMSParams, error)
 	// Describe returns a human-readable description of the changeset.
 	Describe func(in IN, dep DEP) string
 }
 
 // NewFromOnChainSequence creates a Changeset from an operations.Sequence that deploys contracts on-chain and performs write operations.
 // It wraps sequence execution with DataStore and MCMS integration.
 func NewFromOnChainSequence[IN any, DEP any, CFG any](params NewFromOnChainSequenceParams[IN, DEP, CFG]) deployment.ChangeSetV2[CFG] {
```

Comment:
```suggestion
func NewFromOnChainSequence[IN any, DEP any, ChangesetCfg[CFG] any](params NewFromOnChainSequenceParams[IN, DEP, CFG]) deployment.ChangeSetV2[CFG] {
```
Instead of having sequences define `WithMCMS interface`, we just create a ChangesetCfg type that defines the standard MCMS input (currently represented as the MCMSCfg struct) alongside a generic changeset config.